### PR TITLE
Added support for Go 1.15 and removed unnecessary docker images

### PR DESCRIPTION
**Pull request by metamemelord**
Added support for Go 1.15
Removed hello example and init docker images
Added build and test workflows

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      - run: 'docker version'
+      - deploy:
+          command: |
+            if [[ "${CIRCLE_BRANCH}" == "master" && -z "${CIRCLE_PR_REPONAME}" ]]; then
+              git config --global user.email "ci@fnproject.com"
+              git config --global user.name "CI"
+              git branch --set-upstream-to=origin/${CIRCLE_BRANCH} ${CIRCLE_BRANCH}
+
+              # Build images
+              ./build-images.sh 1.15
```
Comment (sudhaku) [inline .circleci/config.yml:57]: Lets use meaningful constants instead of magic numbers.

File: release_images.sh [hunk 2]
```diff
@@ -14,10 +15,10 @@
...
-FROM golang:1.12.7-alpine3.10
-RUN apk add --no-cache git
 
-COPY boilerplate /boilerplate
+user="fnproject"
+image="go"
 
-WORKDIR /boilerplate
-CMD ["/bin/sh", "mod.sh"]
+runtime115="1.15"
```
Comment (sudhaku) [inline release_images.sh:22]: There will be cases when we will need to release more versions than 1.15. So to keep it consistent can we modify this script to take version as an input similar to build script ?

Comment (metamemelord) [reply release_images.sh:22]: This is inherited from fdk-python's way of pushing images: https://github.com/fnproject/fdk-python/blob/master/release_images.sh

for more images, we add new entries to this and build-image shell scripts.

File: .circleci/config.yml (hunk 1 above)
Comment (metamemelord) [reply .circleci/config.yml:57]: This is strictly tied to the FDK directory structure. The version passed here will build dockerfile present in images/build-stage/<version passed here>/Dockerfile. This _**can**_ be made into CircleCI constants, but these values will never be changes.

Comment (sudhaku) [reply .circleci/config.yml:57]: Yes, I understand that. The issue is I will not understand what this value "1.15" means until I look at build-images.sh.

**Review by bgputtan — APPROVED** [review]

**Pull request merged**